### ISSUE TITLE
Generated `main()` passes the wrong arguments to `parser.parse()`

### ISSUE THREAD

**Issue by dnicolodi** [neogeny/TatSu#263]
The `main()` of a generated parser calls the parser `parse()` method like this:
```python
                    return parser.parse(
                        text,
                        rule_name=start,
                        filename=filename,
                        **kwargs
                    )
```
however, the `parse()` method, however the right parameter name is `start` (or `start_rule`, one is kept for backward compatibility, but I don't know which one) and not `rule_name`.

This breaks using the generated parser module as a script, at least in TatSu 5.7.3, but it seems that the current git has the same problem.

In general, it would be nice if the parameter names were validated somewhere.

**Comment by apalala** [comment]
I think this issue is fixed on the `master` branch. Could you check?

**Comment by apalala** [comment]
There are mentions of `rule_name` all over the place, including the docs and tests.

I think that the solution is to change everything to `start`, and add make `PaserConfig` recognize `rule_name` for backwards compatibility.

**Comment by dnicolodi** [comment]
Sorry, I was looking at the sources checked out a the v5.7.3 tag... The bug is indeed not there on git master.

**Comment by apalala** [comment]
The 5.7 branch may have bugfix releases if _someone_ posts the pull request :-)

Yet Python 3.9 just reached its end of bugfix cycle. Why not use Python 3.10?

As main maintainer for TatSu, and with Python 3.11 being published in October this year, I can't find enough time to maintain compatibility with deprecated versions of Python.

The bug is there in the current `master`, but it's mostly a deprecation bug:

```
$ ag -w rule_name
test/parsing_test.py
76:        ast = tatsu.parse(grammar, "test", rule_name='start')
87:        ref_lowercase_result = tatsu.parse(grammar.format(rulename='reflowercase'), test_string, rule_name='start')
88:        ref_uppercase_result = tatsu.parse(grammar.format(rulename='Refuppercase'), test_string, rule_name='start')
90:            result = tatsu.parse(grammar.format(rulename=rulename), test_string, rule_name='start')
93:            result = tatsu.parse(grammar.format(rulename=rulename), test_string, rule_name='start')

tatsu/bootstrap.py
1437:        rule_name=start,

tatsu/codegen/python.py
587:                        rule_name=start,

docs/use.rst
248:    ast = parser.parse('text to parse', rule_name='start')
261:    model = parser.parse(text, rule_name='start', semantics=MySemantics())
273:    model = parser.parse(tokenizer, rule_name='start', semantics=MySemantics())
(tatsu) [master]
```

**Comment by dnicolodi** [comment]
Far from me to question your decisions or ask you to put in extra work, but the most relevant Python libraries aim at supporting all security supported Python releases. In this sense, Python 3.7 is supported till June next year and is far from deprecated. I follow the same principle for my "serious" projects. Supporting the Python version contained in the latest Debian stable and the last two Ubuntu LTS releases seems reasonable.

**Comment by dnicolodi** [comment]
> The bug is there in the current `master`, but it's mostly a deprecation bug:

Do you mean that the remaining mentions of `rule_name` should be removed? I can prepare a patch for this.

**Comment by dnicolodi** [comment]
By the way, which features supported by Python 3.10 only are used in the codebase and where? I just ran the tests on master with Python 3.9 and they all pass.

**Comment by apalala** [comment]
Python 3.9 just had its last bugfix release.

I don't think that there is any >=3.10 only code in TatSu, but it makes sense, as a maintainer, to synchronize with the Python release cycle.

Come October this year, TatSu should be free to use _all_ Python 3.11 features, so the docs should state that compatibility with previous versions of Python is not maintained.

**Comment by dnicolodi** [comment]
Your project and your rules, but requiring python 3.10 when there is no actual dependency to any python 3.10 feature seems to put unnecessary requirements to the users of otherwise a quite fine tool. Also, your definition of currently supported Python versions is different from the one used by most widely used python libraries.

**Comment by apalala** [comment]
Python 3.9 Release Schedule:

https://peps.python.org/pep-0596/

**Comment by dnicolodi** [comment]
Where does it say that Python 3.9 is deprecated? It says that security support is provided till October 2025. However, it is evident that you have your opinion on the support status of Python releases and it will not change. Maybe we can approach this from another perspective: what problem did it solve to restrict the Python version supported by TatSu to Python 3.10 only?

**Comment by dnicolodi** [comment]
> > The bug is there in the current `master`, but it's mostly a deprecation bug:
> 
> Do you mean that the remaining mentions of `rule_name` should be removed? I can prepare a patch for this.

I would also like to go back to the topic of this ticket. What should be the solution for the `start` vs `rule_name` argument name inconsistency?

**Comment by apalala** [comment]
We should settle for `start` and leave backwards compatibility for `rule_name`  in `PaserConfig`.

**Comment by dnicolodi** [comment]
Please have a look at #273 for fixes around the handling of `start` vs `rule_name` and to #270 for a first contribution toward supporting more Python versions.

**Comment by dnicolodi** [comment]
Fixed by #273
